### PR TITLE
Feature decorate PR

### PR DESCRIPTION
**Pull request by codehass**
# OOP school library: decorate a class
Hi ✋ CR
- Create a Nameable class.
- Turn the Person class to Nameable
- Create a Decorator class.
- Create a CapitalizeDecorator class.
- Create a TrimmerDecorator class.

Thank you for reviewing my PR 😀😀

### CODE REVIEW

File: nameable.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError, 'Subclasses should implement this method.'
```
Comment (FrederickMih) [inline nameable.rb]: - [**Optional**] Let's kindly raise the error like this:
`raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"`

File: person.rb [hunk 2]
```diff
@@ -1,8 +1,11 @@
-class Person
+require_relative './nameable'
+
+class Person < Nameable
```
Comment (FrederickMih) [inline person.rb:5]: - [ ] As requested, let's kindly make sure the `Person `class has a method `correct_name` as below

```ruby
class Person < Nameable
  .
  .

    def correct_name
       name
    end
  .
  .
end
```

File: decorator.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require_relative './nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
```
Comment (FrederickMih) [inline decorator.rb:6]: - [ ] It seems you omitted Getters and Setters for nameable, that's

` attr_accessor :nameable`

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand:  @codehass, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!





## Highlights :green_circle: 

- PR is professional :+1: 
- Linter checks passed :heavy_check_mark: 


## Required Changes ♻️

- [ ] @codehass after fixing all the issues below, kindly make sure all the methods are working by testing them.

_Check the comments under the review._

## Optional suggestions
**N/A**
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 4]
```diff
@@ -1,8 +1,11 @@
-class Person
+require_relative './nameable'
+
```
Comment (prolajumokeoni) [inline person.rb:4]:
```person = Person.new(22, 'maximilianus')
   person.correct_name 
   capitalizedPerson = CapitalizeDecorator.new(person)
   capitalizedPerson.correct_name
   capitalizedTrimmedPerson = TrimmerDecorator.new(capitalizedPerson)
   capitalizedTrimmedPerson.correct_name```

[OPTIONAL]
- Kindly import classes needed to run the code in person.rb so there are no errors in  the terminal

**Review by prolajumokeoni — APPROVED** [review]
Hi @codehass 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
✔️ Created a class Nameable
✔️ Person and Decorator class inherits from nameable

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**